src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -43,6 +46,48 @@ public DiagnosticsTelemetryModule()
         }
 
         /// <summary>
+        /// Gets or sets a value indicating whether or not the Health Heartbeat feature is disabled.
+        /// </summary>
+        public bool IsHeartbeatEnabled
+        {
+            get
+            {
+                return this.isHeartbeatEnabled;
```

Comment:
Why not return the actual state of provider? Like `this.HeartbeatProvider.IsEnabled`?

Reply:
Perhaps it was unnecessary for me to do this (indeed, I didn't want to store the value for that in two places).

What about the case where HeartbeatProvider is not yet instantiated (Intitialize has not yet been called) and we want to disable it before then. I think this is the case when we set the value in our ApplicationInsights.config file... no? 

If I am correct, the order is this:
1. DiagnosticsTelemetryModule gets instantiated.
2. The public properties (IsHeartbeatEnabled) gets set based on values in the configuration file. 
3. DiagnosticsTelemetryModule is Initialized